### PR TITLE
notification queue

### PR DESCRIPTION
**Pull request by ngokevin**
Basic gist:

notification

- I updated the copy + duration since that screenshot.
- Not sure why the transition isn't triggering both ways. But not spending much time thinking about it.
- Added messages for submit/fail for addon/versions

### CODE REVIEW

**Comment by ngokevin** [comment]
fun one for you, r? @chuckharmston

**Comment by chuckharmston** [comment]
Weird that the background changed when you selected a file. Just a GIF artifact?

File: src/media/css/components/notification.styl [hunk 1]
```diff
@@ -0,0 +1,21 @@
+@import '../lib';
+
+
+.notification
+  background $color--black
+  border-top 1px solid $color--grey
+  bottom 0
+  color $color--white
```
Comment (chuckharmston) [inline src/media/css/components/notification.styl]: `$notification--var`s?

File: src/media/css/components/progressBar.styl [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@import 'lib';
+@import '../lib';
```
Comment (chuckharmston) [inline src/media/css/components/progressBar.styl]: :car:

File: src/media/js/addon/actions/submit.js [hunk 3]
```diff
@@ -74,6 +76,7 @@ export function submit(fileData) {
         dispatch(pollValidator(res.body.id));
       }, err => {
         dispatch(validationFail(JSON.parse(err.response.text).detail));
+        _errorNotification(dispatch);
```
Comment (chuckharmston) [inline src/media/js/addon/actions/submit.js]: Not a fan of this name.

File: src/media/js/site/actions/notification.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      duration,
+      message
+    }));
+
+    // Shift the notification when it comes time.
+    setTimeout(() => {
+      dispatch(notificationShift());
+    }, totalDuration + duration);
+  };
+}
```
Comment (chuckharmston) [inline src/media/js/site/actions/notification.js]: This is top notch. Nicely done.

File: src/media/js/site/components/notification.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import classnames from 'classnames';
+import React from 'react';
+
+
+export default class Notification extends React.Component {
+  static propTypes = {
+    notification: React.PropTypes.any
```
Comment (chuckharmston) [inline src/media/js/site/components/notification.js]: I think we'd want `PropTypes.node` here, right?

**Comment by chuckharmston** [comment]
This commit is a triumph of you and React. Nicely done. r+

File: src/media/js/site/components/notification.js (hunk 5 above)
Comment (ngokevin) [inline src/media/js/site/components/notification.js]: It could be an JS object or null

File: src/media/js/addon/actions/submit.js (hunk 3 above)
Comment (ngokevin) [inline src/media/js/addon/actions/submit.js]: validationErrorNotification?

File: src/media/css/components/progressBar.styl (hunk 2 above)
Comment (ngokevin) [inline src/media/css/components/progressBar.styl]: sweet wheels

File: src/media/js/addon/actions/submit.js (hunk 3 above)
Comment (chuckharmston) [inline src/media/js/addon/actions/submit.js]: r+ that

File: src/media/css/components/notification.styl (hunk 1 above)
Comment (ngokevin) [inline src/media/css/components/notification.styl]: maybe_not. not re--used, already a $var

**Comment by ngokevin** [comment]
would be cool if there was an animation when once notification is replacing another but not gonna do that here

File: src/media/js/site/components/notification.js (hunk 5 above)
Comment (ngokevin) [inline src/media/js/site/components/notification.js]: well can't really handle a null proptype

**Pull request merged**